### PR TITLE
Adding the ability for couchdb to get a view name

### PR DESCRIPTION
**Pull request by georgethomas111**
@johnhof

### CODE REVIEW

**Comment by georgethomas111** [comment]
@gsquire

File: db.go [hunk 1]
```diff
@@ -92,15 +92,21 @@ func (db *Database) Create() error {
 	return nil
 }
 
-func (db *Database) GetView(docName string, viewName string) (error, []byte) {
+func (db *Database) GetView(docName string, viewName string, key string) (error, []byte) {
```
Comment (gsquire) [inline db.go]: It's not a huge deal, but I typically return errors as a second value in a tuple. This is what Go authors usually do.

File: db.go [hunk 2]
```diff
@@ -92,15 +92,21 @@ func (db *Database) Create() error {
...
-	_, body, _ := db.Req.Get(prefix).End()
+	var body string
+	if key == "" {
+		prefix := docName + "/_view/" + viewName
+		log.Info("Getting view name " + prefix)
+		_, body, _ = db.Req.Get(prefix).End()
+	} else {
+		prefix := docName + "/_view/" + viewName
+		_, body, _ = db.Req.Get(prefix).Query("key=" + key).End()
+	}
```
Comment (gsquire) [inline db.go:114]: Are you ignoring errors here?

File: db.go (hunk 1 above)
Comment (georgethomas111) [inline db.go]: Yup I also saw that after writing a bit of code. Have a number of functions that have it that way. Trying to make sure new ones don't follow this as this function is already used in a lot of places right now.

Comment (georgethomas111) [inline db.go]: I corrected this for the next commit.

File: db.go (hunk 2 above)
Comment (georgethomas111) [inline db.go:114]: Looks like i was ignoring some. No all but some. Added extra handling for it. 

Thanks

**Comment by georgethomas111** [comment]
@gsquire Corrected the changes suggested. Let me know if you have more. Thanks

File: db.go [hunk 3]
```diff
@@ -92,30 +93,43 @@ func (db *Database) Create() error {
 	return nil
 }
 
-func (db *Database) GetView(docName string, viewName string) (error, []byte) {
+func (db *Database) GetView(docName string, viewName string, key string) ([]byte, error) {
+
+	log.Print("couch : GetView key " + key + " in viewName " + viewName + " of desDoc " + docName)
```
Comment (gsquire) [inline db.go]: I think a formatted string would be easier to grok. https://godoc.org/github.com/Sirupsen/logrus#Printf

File: db.go [hunk 4]
```diff
@@ -92,30 +93,43 @@ func (db *Database) Create() error {
...
-	log.Info("Getting view name " + prefix)
-	_, body, _ := db.Req.Get(prefix).End()
+	var body string
+	var errs []error
+	if key == "" {
+		prefix := docName + "/_view/" + viewName
+		log.Info("Getting view name " + prefix)
+		_, body, errs = db.Req.Get(prefix).End()
+	} else {
+		key = "\"" + key + "\""
```
Comment (gsquire) [inline db.go]: You can use a raw string literal here to avoid escaping.

``` go
key = `"` + key + `"`
```

File: db.go [hunk 5]
```diff
@@ -92,30 +93,43 @@ func (db *Database) Create() error {
...
+		prefix := docName + "/_view/" + viewName
+		_, body, errs = db.Req.Get(prefix).Query("key=" + key).End()
+	}
 
+	if len(errs) > 0 {
+		return nil, errors.New("Database : " + fmt.Sprint("%v", errs))
```
Comment (gsquire) [inline db.go]: A more descriptive error would help in the future.

**Comment by georgethomas111** [comment]
@gsquire Corrected the comments. Thanks

**Pull request merged**